packages/@ember/-internals/metal/lib/property_get.ts

```diff
@@ -181,12 +181,22 @@ export function _getPath<T extends object>(root: T, path: string | string[]): an
   @param {Object} defaultValue The value to return if the property value is undefined
   @return {Object} The property value or the defaultValue.
   @public
+  @deprecated
 */
 export function getWithDefault<T extends object, K extends Extract<keyof T, string>>(
   root: T,
   key: K,
   defaultValue: T[K]
 ): T[K] {
+  deprecate(
+    'Using getWithDefault has been deprecated. Instead, consider using Ember get and explicitly checking for undefined.',
+    false,
+    {
+      id: 'ember-metal.get-with-default',
+      until: '4.0.0',
+    }
```

Comment:
We need to add a `url` property that links to the deprecation guide URL

Reply:
Still waiting on this https://github.com/ember-learn/deprecation-app/pull/600 but put the URL that it will generate